### PR TITLE
Enable reshape operator test=develop

### PR DESCRIPTION
**Pull request by mozga-intel**
Enable the reshape operator for the nGraph bridge. BERT model uses it.

### CODE REVIEW

File: paddle/fluid/platform/ngraph_helper.h [hunk 1]
```diff
@@ -189,6 +187,22 @@ inline void TrimTrailingSingularDims(ngraph::Shape* shape) {
     }
   }
 }
+
+ngraph::element::Type GetNgType(paddle::framework::proto::VarType::Type dtype) {
```
Comment (tensor-tang) [inline paddle/fluid/platform/ngraph_helper.h:191]: seems much duplicated code with other PRs ?

https://github.com/PaddlePaddle/Paddle/pull/17437/files#diff-67f22a637cafe74d43dcbf4476c8078eR181

File: paddle/fluid/operators/ngraph/ngraph_engine.cc [hunk 2]
```diff
@@ -134,12 +134,11 @@ static std::vector<std::vector<int>> NgraphOpIntervals(
   int pivot = left;
   while (pivot < right) {
     auto op_type = ops->at(pivot)->Type();
-    if (NgraphBridge::isRegister(op_type)) {
+    if (!NgraphBridge::isSupported(ops->at(pivot))) {
```
Comment (tensor-tang) [inline paddle/fluid/operators/ngraph/ngraph_engine.cc:137]: any use about `isRegister` then?

File: paddle/fluid/platform/ngraph_helper.h (hunk 1 above)
Comment (tensor-tang) [reply paddle/fluid/platform/ngraph_helper.h:191]: maybe will have conflicts.

File: paddle/fluid/operators/ngraph/ngraph_engine.cc (hunk 2 above)
Comment (mozga-intel) [reply paddle/fluid/operators/ngraph/ngraph_engine.cc:137]: We created a modified version of it, which considers and has much more conditions. On the whole, the function isRegister could be left in ngraph bridge file or I will remove it in separate pull-request.

Comment (tensor-tang) [reply paddle/fluid/operators/ngraph/ngraph_engine.cc:137]: ok

File: paddle/fluid/platform/ngraph_helper.h (hunk 1 above)
Comment (tensor-tang) [reply paddle/fluid/platform/ngraph_helper.h:191]: please considerate reuse this function next time.

**Review by tensor-tang — APPROVED** [review]
LGTM

**Pull request merged**